hcsshim.go

```diff
@@ -14,6 +14,7 @@ import (
 //go:generate go run mksyscall_windows.go -output zhcsshim.go hcsshim.go
 
 //sys coTaskMemFree(buffer unsafe.Pointer) = ole32.CoTaskMemFree
+//sys SetCurrentThreadCompartmentId(compartmentId uint32) (hr error) = iphlpapi.SetCurrentThreadCompartmentId
```

Comment:
I think the more interesting thing is that we are changing compartments outside of HCS, HNS, or hcsshim. @msabansal, what does the caller do with this function?